backend/inspirehep/pidstore/models.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import datetime
+
+import structlog
+from invenio_db import db
+from invenio_pidstore.errors import PIDInvalidAction
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from sqlalchemy.exc import IntegrityError, SQLAlchemyError
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy_utils.models import Timestamp
+
+from inspirehep.pidstore.errors import (
+    PidRedirectionMissing,
+    WrongPidTypeRedirection,
+    WrongRedirectionPidStatus,
+)
+
+LOGGER = structlog.getLogger()
+
+
+class InspireRedirect(db.Model, Timestamp):
+    __tablename__ = "inspire_pidstore_redirect"
+    __table_args__ = (
+        db.Index(
+            "ix_inspire_pidstore_redirect_new_record", "new_pid_type", "new_pid_value"
+        ),
+    )
+
+    original_pid_type = db.Column(db.String(6), nullable=False, primary_key=True)
+    original_pid_value = db.Column(db.String(255), nullable=False, primary_key=True)
+
+    new_pid_type = db.Column(db.String(6), nullable=False)
+    new_pid_value = db.Column(db.String(255), nullable=False)
+
+    @property
+    def new_pid(self):
+        return PersistentIdentifier.get(self.new_pid_type, self.new_pid_value)
```

Comment:
if we want to be able to use `InspireRedirect.new_pid` inside filters (to do stuff like `filter(InspireRedirect.new_pid == PersistentIdentifier), this should be a [`hybrid_property`](https://docs.sqlalchemy.org/en/13/orm/extensions/hybrid.html), but we probably don't need it.

Reply:
Yeah, I also thought about this but I didn't found an use-case for it. So I left it like it is. Changing it to hybrid-property is simple.

Reply:
Actually now I added proper relations and I updated this model. It should be better more readable and more useful now.